tools/mason/MasonTest.chpl

```diff
@@ -90,28 +90,45 @@ private proc runTests(show: bool, run: bool, parallel: bool) {
     // Check for tests to run
     if numTests > 0 {
 
-      coforall test in testNames {
+      var resultDomain: domain(string);
+      var testResults: [resultDomain] string;
+      var numPassed = 0;
+
+      forall test in testNames with (+ reduce numPassed) {
 
         const testPath = "".join(projectHome, '/test/', test);
-        const testName = basename(test.strip(".chpl"));
+        const testName = basename(stripExt(test, ".chpl"));
 
         // get the string of dependencies for compilation
         const compopts = getDependencyString(sourceList, testName);
 
         const moveTo = "-o " + projectHome + "/target/test/" + testName;
         const compCommand = " ".join("chpl",testPath, projectPath, moveTo, compopts);
         const compilation = runWithStatus(compCommand);
-
         if show then writeln(compCommand);
+
         if compilation != 0 {
           writeln("compilation failed for " + test);
         }
         else {
           if show || !run then writeln("compiled ", test, " successfully");
+          if parallel {
+            var result = runTestBinary(projectHome, testName, show);
+            if result != 0 {
+              testResults[testName] = "Failed";
+            }
+            else {
+              testResults[testName] = "Passed";
+              numPassed += 1;
+            }
+          }
         }
       }
-      if run {
-        runTestBinaries(projectHome, testNames, numTests, show, parallel);
+      if run && !parallel {
+        runTestBinaries(projectHome, testNames, numTests, show);
+      }
+      else if run && parallel {
+        printTestResults(testResults, numTests, numPassed, show);
```

Comment:
Why not pull the `printTestResults` out of `runTestBinaries` so that this is just:

```chpl
else if run then printTestResult(..);
```